math/factorial_recursion.c

```diff
@@ -0,0 +1,33 @@
+#include <stdio.h>

```

Comment:
please add a multi-line comment explaining the formation of this algorithm its uses and author as well as references to the sources.